File: Variable/Python/variable.py
# Primitive data types
age = 30                    # Integer
pi = 3.14                   # Float
name = "John Doe"           # String
greeting = 'Hello, World!'  # String
is_valid = True             # Boolean
is_active = False           # Boolean
empty_value = None          # NoneType

# List (dynamic array)
numbers = [1, 2, 3, 4]
numbers.append(5)  # Adding an element to the list

# Tuple (immutable collection)
coordinates = (10.0, 20.0)

# Dictionary (key-value pairs)
person = {
    "name": "John Doe",
    "age": 30
}

# Set (unordered, unique elements)
unique_numbers = {1, 2, 3, 4}
unique_numbers.add(5)  # Adding a unique element

# Function
def greet(name):
    return f"Hello, {name}!"

# Class and Object
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

john = Person("John Doe", 30)

# Printing values
print("Age:", age)                   # Output: Age: 30
print("Pi:", pi)                     # Output: Pi: 3.14
print("Name:", name)                 # Output: Name: John Doe
print("Greeting:", greeting)         # Output: Hello, World!
print("Is Valid:", is_valid)         # Output: Is Valid: True
print("Is Active:", is_active)       # Output: Is Active: False
print("Empty Value:", empty_value)   # Output: Empty Value: None

# Working with lists
print("Numbers:", numbers)           # Output: Numbers: [1, 2, 3, 4, 5]

# Working with tuples
print("Coordinates:", coordinates)   # Output: Coordinates: (10.0, 20.0)

# Working with dictionaries
print("Person:", person)             # Output: Person: {'name': 'John Doe', 'age': 30}

# Working with sets
print("Unique Numbers:", unique_numbers) # Output: Unique Numbers: {1, 2, 3, 4, 5}

# Using functions
print(greet("Alice"))                # Output: Hello, Alice!

# Using classes and objects
print(f"Person Name: {john.name}, Age: {john.age}") # Output: Person Name: John Doe, Age: 30

File: Variable/Javascript/variable.js
// Primitive data types
let age = 30; // Number
const pi = 3.14; // Constant number
let name = "John Doe"; // String
let greeting = "Hello, World!"; // String
let isValid = true; // Boolean
let notAssigned; // Undefined
let empty = null; // Null
let uniqueId = Symbol("id"); // Symbol

// Object
let person = {
  name: "John Doe",
  age: 30,
  greet: function () {
    return `Hello, my name is ${this.name}`;
  },
};

// Array
let numbers = [1, 2, 3, 4];
numbers.push(5); // Adding an element to the array

// Function
function greet(name) {
  return `Hello, ${name}!`;
}

// BigInt
let largeNumber = BigInt("123456789012345678901234567890");

// Map
let myMap = new Map();
myMap.set("one", 1);
myMap.set("two", 2);

// Set
let mySet = new Set([1, 2, 3]);
mySet.add(4); // Adding an element to the set

// Logging values
console.log("Age:", age); // Output: Age: 30
console.log("Pi:", pi); // Output: Pi: 3.14
console.log("Name:", name); // Output: Name: John Doe
console.log("Greeting:", greeting); // Output: Hello, World!
console.log("Is Valid:", isValid); // Output: Is Valid: true
console.log("Not Assigned:", notAssigned); // Output: Not Assigned: undefined
console.log("Empty:", empty); // Output: Empty: null
console.log("Unique ID:", uniqueId); // Output: Unique ID: Symbol(id)

// Using objects
console.log("Person:", person); // Output: Person: { name: 'John Doe', age: 30, greet: [Function: greet] }
console.log(person.greet()); // Output: Hello, my name is John Doe

// Using arrays
console.log("Numbers:", numbers); // Output: Numbers: [ 1, 2, 3, 4, 5 ]

// Using functions
console.log(greet("Alice")); // Output: Hello, Alice!

// Using BigInt
console.log("Large Number:", largeNumber); // Output: Large Number: 123456789012345678901234567890n

// Using maps
console.log("Map:", myMap); // Output: Map: Map { 'one' => 1, 'two' => 2 }

// Using sets
console.log("Set:", mySet); // Output: Set: Set { 1, 2, 3, 4 }
